fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CreateTableAnalyzer.java

```diff
@@ -0,0 +1,365 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+package com.starrocks.sql.analyzer;
+
+import com.google.common.base.Strings;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.starrocks.analysis.ColumnDef;
+import com.starrocks.analysis.CreateTableStmt;
+import com.starrocks.analysis.DistributionDesc;
+import com.starrocks.analysis.IndexDef;
+import com.starrocks.analysis.KeysDesc;
+import com.starrocks.analysis.PartitionDesc;
+import com.starrocks.analysis.RangePartitionDesc;
+import com.starrocks.analysis.TableName;
+import com.starrocks.catalog.AggregateType;
+import com.starrocks.catalog.Catalog;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Index;
+import com.starrocks.catalog.KeysType;
+import com.starrocks.catalog.PartitionType;
+import com.starrocks.catalog.PrimitiveType;
+import com.starrocks.cluster.ClusterNamespace;
+import com.starrocks.common.AnalysisException;
+import com.starrocks.common.Config;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
+import com.starrocks.common.FeConstants;
+import com.starrocks.common.FeNameFormat;
+import com.starrocks.external.elasticsearch.EsUtil;
+import com.starrocks.mysql.privilege.PrivPredicate;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.analyzer.relation.Relation;
+import org.apache.commons.collections.CollectionUtils;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+import static com.starrocks.catalog.AggregateType.BITMAP_UNION;
+
+public class CreateTableAnalyzer {
+
+    private final Catalog catalog;
+    private final ConnectContext session;
+    private static final Set<String> engineNames;
+
+    static {
+        engineNames = Sets.newHashSet();
+        engineNames.add("olap");
+        engineNames.add("mysql");
+        engineNames.add("elasticsearch");
+        engineNames.add("hive");
+    }
+
+    public CreateTableAnalyzer(Catalog catalog, ConnectContext session) {
+        this.catalog = catalog;
+        this.session = session;
+    }
+
+    public Relation transformCreateTableStmt(CreateTableStmt createTableStmt) {
+        createTableStmt.setClusterName(session.getClusterName());
+        TableName tableName = createTableStmt.getDbTbl();
+        String tableNameStr = tableName.getTbl();
+        analyzeTableName(tableName);
+
+        if (Strings.isNullOrEmpty(tableNameStr) || !tableNameStr.matches(FeNameFormat.COMMON_NAME_REGEX)) {
```

Comment:
`FeNameFormat.checkTableName(tableNameStr);`

Reply:
checkTableName throws AnalysisException and it must be catch so take out the code inside here.

Reply:
I think it's better to call the public method (deal with the exception here).